src/components/common/Menu/Menu.jsx

```diff
@@ -0,0 +1,220 @@
+import React, { useState, useEffect } from "react";
+import "./styles.css";
+import {
+  AppBar,
+  Toolbar,
+  Drawer,
+  Container,
+  IconButton,
+  TextField,
+  Typography,
+  Box,
+  makeStyles,
+  Button,
+  Grid,
+} from "@material-ui/core";
+import { Close as CloseIcon, Menu as MenuIcon } from "@material-ui/icons";
+import StyledSwitch from "../Switch/StyledSwitch";
+import parseQueryVariables from "../../../helpers/queryParser";
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    backgroundColor: "#1a1c2b",
+    color: "white",
+    width: "60%",
+    [theme.breakpoints.up("lg")]: { width: "20%" },
+    [theme.breakpoints.only("md")]: { width: "30%" },
+    [theme.breakpoints.only("sm")]: { width: "40%" },
+    [theme.breakpoints.only("xs")]: { width: "60%" },
+  },
+  input: {
+    backgroundColor: "white",
+    color: "#1a1c2b",
+    borderRadius: "15px",
+    padding: "9px",
+    marginRight: "10%",
+    marginTop: "1.5%",
+    width: "100%",
+  },
+}));
+
+const Menu = (props) => {
+  const classes = useStyles();
+  const accessGps = props.accessGps;
+  const showPosition = props.showPosition;
+
+  const [open, setState] = useState(false);
+  const [longitude, setLongitude] = useState(props.coordinate.lon);
+  const [latitude, setLatitude] = useState(props.coordinate.lat);
+  const [isManual, setSwitchState] = useState(false);
+
+  useEffect(() => {
+    if (!isManual) {
+      setLongitude(props.coordinate.lon);
+      setLatitude(props.coordinate.lat);
+    }
+  }, [isManual, props.coordinate]);
+
+  useEffect(() => {
+    const params = parseQueryVariables(window.location.search);
+    if (Object.keys(params).length === 2) {
+      setSwitchState(true);
+      setLatitude(params.latitude);
+      setLongitude(params.longitude);
+    }
+  }, []);
+
+  const toggleDrawer = (open) => (event) => {
+    setState(open);
+  };
+
+  const handleChange = () => (event) => {
+    setSwitchState(event.target.checked);
+  };
+
+  const handleLatChange = (event) => {
+    setLatitude(event.target.value);
+  };
+
+  const handleLonChange = (event) => {
+    setLongitude(event.target.value);
+  };
+
+  const handleManualSubmit = () => {
+    const position = {
+      coords: { latitude: Number(latitude), longitude: Number(longitude) },
+    };
+    showPosition(position);
+  };
+
+  const handleRefreshGPS = () => {
+    accessGps();
+  };
+
+  return (
+    <div className="menu">
+      <AppBar position="static" style={{ background: "#202333" }}>
+        <Container maxWidth="lg" disableGutters={true}>
+          <Toolbar>
+            <IconButton
+              aria-label="open drawer"
+              onClick={toggleDrawer(true)}
+              style={{ margin: "-17px", color: "white" }}
+            >
+              <MenuIcon />
+            </IconButton>
+
+            <Drawer
+              anchor="right"
+              open={open}
+              onClose={toggleDrawer(false)}
+              onOpen={toggleDrawer(true)}
```

Comment:
When we open the drawer we got an error because this property doesn't exists on the component, it can be safely removed.

Reply:
Just fixed as well